provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -905,6 +905,26 @@ func generateSchema() schema.PackageSpec {
 							"  - settings.kubernetes.cluster-dns-ip\n\n" +
 							"For an overview of the available settings, see https://bottlerocket.dev/en/os/1.20.x/api/settings/.",
 					},
+					"userData": {
+						TypeSpec: schema.TypeSpec{Type: "string"},
+						Description: "User specified code to run on node startup. This is expected to handle " +
+							"the full AWS EKS node bootstrapping. If omitted, the provider will configure the user data.\n\n" +
+							"See for more details: https://docs.aws.amazon.com/eks/latest/userguide/launch-templates.html#launch-template-user-data.",
+					},
+					"gpu": {
+						TypeSpec: schema.TypeSpec{Type: "boolean"},
```

Comment:
Would this be any better with defining the explicit default in the schema?

Reply:
We need to know whether it's false or not defined. There's certain combinations that do not work (e.g. you cannot choose an AMI with GPU and then set GPU to false).
Is my understanding wrong that we'd receive `false` in the provider in that case?